frame/support/procedural/src/storage/instance_trait.rs

```diff
@@ -70,42 +69,40 @@ pub fn decl_and_impl(scrate: &TokenStream, def: &DeclStorageDefExt) -> TokenStre
 		.and_then(|i| i.instance_default.as_ref())
 	{
 		impls.extend(quote! {
+			/// Hidden instance generated to be internally used when module is used without
+			/// instance.
 			#[doc(hidden)]
 			pub type #inherent_instance = #default_instance;
 		});
 	} else {
 		let instance_def = InstanceDef {
 			prefix: String::new(),
 			instance_struct: inherent_instance,
-			doc: quote!(#[doc(hidden)]),
+			doc: quote!(
+				/// Hidden instance generated to be internally used when module is used without
+				/// instance.
+				#[doc(hidden)]
+			),
 		};
 		impls.extend(create_and_impl_instance_struct(scrate, &instance_def, def));
 	}
 
 	impls
 }
 
-fn create_instance_trait(
+fn reexport_instance_trait(
+	scrate: &TokenStream,
 	def: &DeclStorageDefExt,
 ) -> TokenStream {
-	let instance_trait = def.module_instance.as_ref().map(|i| i.instance_trait.clone())
-		.unwrap_or_else(|| syn::Ident::new(DEFAULT_INSTANTIABLE_TRAIT_NAME, Span::call_site()));
-
-	let optional_hide = if def.module_instance.is_some() {
-		quote!()
+	if let Some(i) = def.module_instance.as_ref() {
+		let instance_trait = &i.instance_trait;
+		quote!(
+			/// Reexport from support::traits.
+			// This reexport is not strictly needed but used in order not to have breaking change.
+			pub use #scrate::traits::Instance as #instance_trait;
+		)
 	} else {
-		quote!(#[doc(hidden)])
-	};
-
-	quote! {
-		/// Tag a type as an instance of a module.
-		///
-		/// Defines storage prefixes, they must be unique.
-		#optional_hide
-		pub trait #instance_trait: 'static {
-			/// The prefix used by any storage entry of an instance.
-			const PREFIX: &'static str;
-		}
```

Comment:
note: __GeneratedInstantiable is no longer generated because it was supposed to be only internally used (and so all our internal use now use frame_support::traits::Instance directly)